### ISSUE TITLE
Some issues with the VecN library and compatibility with Vec2

### ISSUE THREAD

**Issue by TheOneKevin** [TheOneKevin/JavaPhysicsSimulation#7]
Sometimes you want to be able to do:
`a.add(b).scalarMultiply(c).subtract(d)`
- [ ] See sample resolution [here](https://github.com/TheOneKevin/JavaPhysicsSimulation/commit/9ad6aed3f1e09d3c878feab2f5f5c69455d11471#diff-6ee80e8a3dd9048957fda9bb3c23043cR247).

Also, to reduce the amount of objects created:
- [ ] Subtract is kinda missing (for now, `a.add(VecN.negate(b))` is kinda not great haha)
- [ ] `VecN.set()` needs an option to take in a `VecN`

Not urgent. Take your time. Thanks!

**Comment by Nova-Terra** [comment]
Quick question: T extends VecN essentially means the subclasses of VecN right?

**Comment by TheOneKevin** [comment]
Yeah. I'm also worried the vector libraries might be creating too many objects because they're going to be run many many times per frame 😟
